### PR TITLE
Remove polyfill firrtl code that was moved to firrtl-interpreter

### PR DESCRIPTION
**Pull request by jackkoenig**
These files were moved to firrtl-interpreter in https://github.com/freechipsproject/firrtl-interpreter/pull/222. We need to avoid classpath collisions which could lead to runtime linkage errors.

### CODE REVIEW

**Comment by jackkoenig** [comment]
I think this failed because I pushed it before `firrtl-interpreter` with these APIs had been published yet.

**Comment by jackkoenig** [comment]
Okay so the issue is that both `firrtl` and `firrtl-interpreter` both implement `DriverCompatibility` so there's a classpath collision. This same issue exists currently in this repo but the winning version of that object is different. This isn't great so I'm going to resolve the conflict in `firrtl-interpreter` first.

File: src/main/scala/chisel3/iotesters/DriverCompatibility.scala [hunk 1]
```diff
@@ -10,6 +10,7 @@ import firrtl.options.Viewer.view
 import firrtl.options.phases.DeletedWrapper
 import firrtl.options.{Dependency, OptionsException, OptionsView, OutputAnnotationFileAnnotation, Phase, PhaseManager, StageError, Unserializable}
 import firrtl.stage.phases.DriverCompatibility.TopNameAnnotation
+import firrtl.stage.phases.DriverCompatibilityExtensions
```
Comment (jackkoenig) [inline src/main/scala/chisel3/iotesters/DriverCompatibility.scala:13]: This is unfortunately needed. Provided by https://github.com/freechipsproject/firrtl-interpreter/pull/223

**Review by chick — APPROVED** [review]
Looks good

**Pull request merged**